front/components/app/NewBlock.jsx

```diff
@@ -1,39 +1,81 @@
 import { PlusIcon } from "@heroicons/react/20/solid";
 import { classNames } from "../../lib/utils";
-import { Menu, Transition } from "@headlessui/react";
+import { Popover, Transition } from "@headlessui/react";
 import { Fragment } from "react";
+import { usePopper } from "react-popper";
+import { useState } from "react";
 
 export default function NewBlock({ spec, disabled, onClick, direction }) {
+  let [referenceElement, setReferenceElement] = useState();
+  let [popperElement, setPopperElement] = useState();
+  let { styles, attributes } = usePopper(referenceElement, popperElement, {
+    placement: direction === "up" ? "top-end" : "bottom-end",
+  });
+
   let containsInput = spec.filter((block) => block.type == "input").length > 0;
   let blocks = [
-    { type: "data", display: ["data"] },
-    { type: "llm", display: ["llm"] },
-    { type: "code", display: ["code"] },
-    { type: "search", display: ["search"] },
-    { type: "map_reduce", display: ["map", "reduce"] },
+    {
+      type: "llm",
+      typeNames: ["llm"],
+      name: "Large Language Model (LLM)",
+      description:
+        "Ask an artificial intelligence Large Language Model to complete a prompt for you. Current LLMs supported: OpenAI's GPT-3 and Cohere.",
+    },
+    {
+      type: "data",
+      typeNames: ["data"],
+      name: "Data",
+      description:
+        "Load a dataset to be used for every run of the Dust app. Typically used to seed a few-shot prompt to an LLM block.",
+    },
+    {
+      type: "code",
+      typeNames: ["code"],
+      name: "JavaScript",
+      description:
+        "Run a snippet of JavaScript to modify, augment, or combine results from other blocks.",
+    },
+    {
+      type: "search",
+      typeNames: ["search"],
+      name: "Google Search",
+      description:
+        "Issue a query to Google so you can feed the results to other blocks.",
+    },
+    {
+      type: "map_reduce",
+      typeNames: ["map", "reduce"],
+      name: "Map Reduce",
+      description:
+        "Map over an array and execute a sequence of blocks in parallel.",
+    },
   ];
   if (!containsInput) {
-    blocks.splice(0, 0, { type: "input", display: ["input"] });
+    blocks.splice(0, 0, {
+      type: "input",
+      typeNames: ["input"],
+      name: "Input",
+      description:
+        "Select a dataset that is used as the input to this Dust app, like the arguments to a function. Each element in the dataset kicks off a separate parallel run of the Dust app.",
```

Comment:
Select a dataset of inputs used for the design your Dust app. Each element in the dataset kicks off a separate parallel execution of the Dust app.